src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -383,10 +407,55 @@ def bulk_set(self, flags):
         data = set(flags)
         added = data - enabled_flags
         removed = enabled_flags - data
-        for flag in added:
-            CephService.send_command('mon', 'osd set', '', key=flag)
-        for flag in removed:
-            CephService.send_command('mon', 'osd unset', '', key=flag)
+
+        self._update_flags('set', added)
+        self._update_flags('unset', removed)
+
         logger.info('Changed OSD flags: added=%s removed=%s', added, removed)
 
         return sorted(enabled_flags - removed | added)
+
+    @Endpoint('PUT', 'individual', query_params=['flags', 'ids'])
+    @UpdatePermission
+    @EndpointDoc('Sets OSD flags for a subset of individual OSDs.',
+                 parameters={
+                     'flags': ({'noout': (bool, 'Sets/unsets `noout`', True, None),
+                                'noin': (bool, 'Sets/unsets `noin`', True, None),
+                                'noup': (bool, 'Sets/unsets `noup`', True, None),
+                                'nodown': (bool, 'Sets/unsets `nodown`', True, None)},
+                               'Directory of flags to set or unset. The flags '
+                               '`noin`, `noout`, `noup` and `nodown` are going to '
+                               'be considered only.'),
+                     'ids': ([str], 'List of OSD ids the flags should be applied '
+                                    'to.')
```

Comment:
According to [Ceph docs](https://docs.ceph.com/docs/master/rados/operations/health-checks/#osd-flags), `id` (or `<who>` in Ceph terminology) could be any CRUSH entity (nodes, class devices...):
```sh
ceph osd set-group noup,noout host-foo
```
We are allowing that, as we don't validate the input here. Is this desired behaviour?

I checked code injection (by passing `id = "1; ceph osd rm osd.1"`) and it seems Ceph catches it.

Reply:
I added `assert (all(id.isdigit() for id in ids))` to the endpoint implementation. Thanks!

Reply:
Thanks!